packages/webapp/src/containers/Animals/SingleAnimalView/index.tsx

```diff
@@ -149,45 +164,79 @@ function SingleAnimalView({ isCompactSideMenu, history, match, location }: AddAn
       console.error(e);
       dispatch(enqueueErrorSnackbar(t('message:ANIMALS.FAILED_UPDATE_BATCH')));
     }
+  };
+
+  const getInventoryId = () => {
+    if (!selectedAnimal && !selectedBatch) {
+      return '';
+    }
 
-    onGoForward();
+    const animalOrBatch = AnimalOrBatchKeys[selectedAnimal ? 'ANIMAL' : 'BATCH'];
+    return generateInventoryId(animalOrBatch, (selectedAnimal || selectedBatch)!);
+  };
+
+  const { onConfirmRemoveAnimals, removalModalOpen, setRemovalModalOpen } = useAnimalOrBatchRemoval(
+    [getInventoryId()],
```

Comment:
Perhaps we could modify `useAnimalOrBatchRemoval` to receive the actual entity IDs and whether they're a batch or animal instead of the inventory IDs? Seems more straight forward since here we're generating the inventory ID just to pass it to the hook, and then the hook has to re-parse the actual IDs from it

Reply:
I hesitated to refactor since it would involve changes to an unrelated file, but it’s true that I wasn’t happy with the inefficiency. Let me open another PR for this!